### PR TITLE
Refactoring Key Validation Based on dictionary #95

### LINKED ISSUE #95: Refactoring key validation in a dictionary body
### I'm submitting a 
- [x] Enhancement Request.

### Current Behaviour:

We are using try catch to check every key in the dictionary and assigning it to new variable.
rather to validate the current payload we should have a generic function in which we can pass the key and value to check and it will return the keys or exception from that function 
[Current Sample code structure] (https://ibb.co/ciM4Vn)

### Do you want to work on this issue ?

yes
@Mec-iS -can you provide any suggestion here

### PR DESCRIPTION
**Pull request by anubhav253**
Fixes #95 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### CODE REVIEW

**Comment by alohamora** [comment]
@anubhav253 I think that the `errorMapping` function is redundant as all the error strings have a similar format, so we can simply print the error string as:
`raise SyntaxError("The " + bodyType + " must have [" + key + "]")`
and pass the respective string in bodyType.

**Comment by anubhav253** [comment]
@alohamora Yeah but I don't wanted to hamper the existing output as my target to refractor the code repetation and using dictionary here also to change the error message we can have a discussion with Mec-iS as I haven't had a talk regarding this.

**Comment by Mec-iS** [comment]
Some basic things, wait for review: 
1. the code should be formatted as specified by PEP-8, check #79 
2. the standard way of formatting string is by using `str.format()` [doc](https://docs.python.org/3.5/library/stdtypes.html#str.format), check also [syntax](https://docs.python.org/3.4/library/string.html#format-string-syntax)

File: hydrus/hydraspec/doc_maker.py [hunk 1]
```diff
@@ -16,56 +41,46 @@ def createDoc(doc: Dict[str, Any], HYDRUS_SERVER_URL: str=None, API_NAME: str=No
         raise SyntaxError("The API Documentation must have [@id]")
 
     # Extract base_url, entrypoint and API name
-    matchObj = re.match(r'(.*)://(.*)/(.*)/vocab#?', id_, re.M | re.I)
-    if matchObj:
-        base_url = matchObj.group(1) + '://' + matchObj.group(2) + '/'
-        entrypoint = matchObj.group(3)
+    match_obj = re.match(r'(.*)://(.*)/(.*)/vocab#?', id_, re.M | re.I)
+    if match_obj:
+        base_url = match_obj.group(1) + '://' + match_obj.group(2) + '/'
```
Comment (Mec-iS) [inline hydrus/hydraspec/doc_maker.py]: String concatenation with `+` should be a rare occurrence. Wherever possible use string formatting `str.format` (rarely concatenation is preferred for performance reason).

Comment (anubhav253) [reply hydrus/hydraspec/doc_maker.py]: @Mec-iS Its updated as per your suggestion.

**Comment by anubhav253** [comment]
@xadahiya Type annotations are added and also updated the branch with develop.

**Comment by anubhav253** [comment]
@xadahiya I have done the changes requested here please review it once.

**Comment by xadahiya** [comment]
@chrizandr  ^^

**Pull request merged**